streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -1580,6 +1580,98 @@ public void shouldThrowStreamsExceptionWhenKeySerializationFailedAndProductionEx
         }
     }
 
+    @Test
+    public void shouldNotFailIfContextIsNotAvailableOnSerializationError() {
+        try (final ErrorStringSerializer errorSerializer = new ErrorStringSerializer()) {
+            final RecordCollector collector = new RecordCollectorImpl(
+                logContext,
+                taskId,
+                streamsProducer,
+                productionExceptionHandler,
+                streamsMetrics,
+                topology
+            );
+
+            assertThrows(
+                StreamsException.class, // should not crash with NullPointerException
+                () -> collector.send(
+                    topic,
+                    "key",
+                    "val",
+                    null,
+                    0,
+                    null,
+                    errorSerializer,
+                    stringSerializer,
+                    sinkNodeName,
+                    null // pass `null` context for testing
```

Comment:
I disagree -- would just raise the question, "why not available" when reading the code, and does not really explain that pass in `null` here is the actual test condition.

Reply:
Do you think "pass `null` context for testing" is clearer?

The test name is `shouldNotFailIfContextIsNotAvailableOnSerializationError`, I think it is quite clear why the context is not available and that this is the test condition.

Reply:
Yes, I think it's clearer :) (otherwise, I would have done it differently). It highlights where the `null` the test name indicates already is exactly passed.

Of course, using `nullContext = null` and pass `nullContext` would also do the trick. It's just that both are equally good for this case IMHO. -- There are other cases for which I would prefer a variable, eg, `UNKNOWN = -1`, and pass `UNKNOWN` instead of `-1` with a comment. However, for the unknown case, I don't care how unknowns is encoded, ie, it could also be `-2` and thus using a variable is better. For `null` vs `nullContext` there is just no read difference, because `null` can only be `null`.

But I guess we are bikeshedding now.